### ISSUE TITLE
dwc:recordNumber: import, store, export

### ISSUE THREAD

**Issue by camwebb** [SpeciesFileGroup/taxonworks#4016]
### Feature or enhancement

A key identifier for plant collections is the collector's field number, usually sequential over a lifetime. Within TW it is not clear where to store this information. In the past I have discussed with TW developers solutions that included creating a new identifier namespace for each collector, but this is too cumbersome. At the moment we are using an attribute.

DwC recommends this information is stored as [recordNumber](https://dwc.tdwg.org/list/#dwc_recordNumber), but this field seems not to be connected to TW. On import the `recordNumber` field is lost. On export, our attribute will not export to `recordNumber`, but only to `TW:DataAttribute:CollectionObject:...`

It would be great to find a way to i) handle this core information more easily in TW, and ii) connect this with the dwc:recordNumber field.

Thanks

### Location

https://sfg.taxonworks.org/collection_objects/XXXX

### Your role

User and developer of other biodiversity tools

**Comment by mjy** [comment]
Largely a duplicate of #3800

**Comment by camwebb** [comment]
Not sure this is the same issue. `dwc:fieldNumber` is a property of an event. `dwc:recordNumber` is a property of an occurrence (=collection object). I searched all the TW Issues for 'recordNumber' and didn't find one that matches our need to use and export recordNumber. Thanks

**Comment by mjy** [comment]
@camwebb Do Botanists use  `dwc:catalogNumber`, and can you provide an example of how it might be different?

**Comment by debpaul** [comment]
> @camwebb Do Botanists use `dwc:catalogNumber`, and can you provide an example of how it might be different?

@mjy yes, botanists use dwc:catalogNumber. What they put in it, varies considerably. Easy to see at iDigBio. Search: limit to country = USA; kingdom = Plantae; add column = catalogNumber to visualize the contents of this field for 58 million plant records. https://www.idigbio.org/portal/search

**Samples:**
UAM023296
UAM:Herb:235288
UAMb:Herb:20716
jtl-sv00047
153108
V0435561F
US 1594134
barcode-02162415
SEL25025
WAG.1370020
S17-43230
BRIT501793
U.1279631
MEL 0280305A
LSU00090756
01098466
RBtecido01419282
RBdna00847515
CONN00059728
V0116812F
ASU0317191
YPM PB 028879
YPM PB 028880.B
YPM PB 028880.A
TENN-B-0118235
S-ple-13751
UTEP:Herb:23346
AMNH_BEE 00075252
USNM PAL411215


FWIW, a few years ago, I asked Tim Robertson at GBIF to run a GBIF query for me, that would reveal the different "types" (patterns) of the contents of this field. I have those results ... somewhere in my old email pile. 

**dwc:fieldNumber** example: RV Sol 87-03-08
and I personally, don't know of many (any?) botanists that use this field. Cam may know different.

See iDigBio, kingdom = Plantae, show values for fieldNumber
![image](https://github.com/user-attachments/assets/c335352d-b28c-4bcb-a8af-4aae23d7bf15)

**dwc:recordNumber** http://rs.tdwg.org/dwc/iri/recordNumber
MANY botanists use this. It's what we're taught. It's the number we personally give to that plant, in our field notebooks. There are many flavors of this. Initials+date+number, perhaps just initials+incremental number, and others.

![image](https://github.com/user-attachments/assets/eb942e66-d31a-4983-9782-3dba7652c334)

**Comment by mjy** [comment]
@debpaul I gathered that it was used.  The question is why?  In the best case scenariou are you saying that *trained* botanist would never use `dwc:catalogNumber`?

**Comment by debpaul** [comment]
> @debpaul I gathered that it was used. The question is why? In the best case scenariou are you saying that _trained_ botanist would never use `dwc:catalogNumber`?

I'm saying the exact opposite. They would "never" use dwc:fieldNumber

**Comment by mjy** [comment]
This is about `dwc:recordNumber` (not `dwc:fieldNumber`) vs. `dwc:catalogNumber`.

**Comment by debpaul** [comment]
> This is about `dwc:recordNumber` (not `dwc:fieldNumber`) vs. `dwc:catalogNumber`.

Yup, I'm attempting to show clearly the difference. dwc:catalogNumbers are generated by the collections staff, NOT the collector. The "dwc:recordNumber" is exactly what botanists use to identify their own collected items. (At the iDigBio site, we "labeled" this "CollectorNumber" b/c folks didn't understand "recordNumber" as a term). In the last screenshot above, you can see, no relationship between the value in the **catalogNumber** field and the **CollectorNumber** field (aka:dwc:recordNumber).

**Comment by mjy** [comment]
For the record I feel that `catalogNumber` and `recordNumber` are semantically blurred.  While we *are* going to split them out in TaxonWorks to facilitate their intdependant use per TDWG it sort of feels like we're adopting standards and forgetting what might become a change to `better` practices.  Maybe.

Identifiers should identify, they shouldn't imply *who* made the identifier.  If we want to ensure the "who" then we need standards that allow us to include the who as a proper attribute.  I.e. what good is recordNumber `NS123` vs. catalogNumber `NS123` without reference (field being filled in) to the who?  At least TaxonWorks addresses this through its ability to create explicit Namespaces (so maybe this isn't an issue in TW).  However,  in TW both require some Namespace, so they are semantically the same in TW except what we mentally understand, hinting that they need to be conceptually merged at a broader level, or standards need to be better refined to minimize the inferrences they imply.

Closing this for #3800 where the aggregate identifier work is being done.

**Comment by camwebb** [comment]
> For the record I feel that catalogNumber and recordNumber are semantically blurred. 

Agree

> Identifiers should identify, they shouldn't imply who made the identifier. If we want to ensure the "who" then we need standards that allow us to include the who as a proper attribute.

Agree, again. Arctos has an IssuedBy field for all identifiers that links to an Agent.

Having a dwc:recordNumber is important, simply for cross compatibility. Thanks for that. Exporting the collector(s) and dwc:recordNumber will provide the collector number identifier with its IssuedBy. 

The issue of the UI is different. We cobbled together a faux-identifier so that we could use `Collector space CollectorNumber` as a label. If there were a specific 'label' field which was used by the UI to give a name to the collection object, independent of the identifiers and their order, that would be a better solution all round. It might use an identifier for the label in the case of no label having been specified. Just a thought.

I know the 'best' current solution is to create a collection code namespace for every collector, but the time/work required to do this for hundreds/thousands of collectors is prohibitive.

**Comment by mjy** [comment]
@camwebb We're working to add a panel for RecordNumber to Comprehensive (users can hide/show as needed) and CE creator.  You'll still be required to create a Namespace for the Identifiers, you can create a generic one specific to your needs, some thing like "Botantical RecordNumber (Webb)", the values will be the full string.  Since you can lock the choice, or pin the Namespace and apply quickly, there shouldn't be too much burden to locking into its use.

I've also enforced that the FieldNumber is identical to the `verbatim_trip_identifier` CE attribute.  To make this test pass your Namespace here (and above) needs to have the "virtual" attribute assigned, this means the short-name is not appended to the identifier value.  If it turns out that verbatim use really is different than this we'll re-vist that constraint, but I want to see it used "in the wild" for a bit.

**Comment by mjy** [comment]
Reading this open tab I should clarify, just in case.  TW best practice will still require a Namespace, however in your situation where you don't want to point to indivual collector's Namespaces, you can create a single universal one, and just use that for all, the "Webb" would just let you know which one to select.  You'll select "virtual" for that Namespace so that short name of it is not appended to the full text of the RecordNumber that you place in `identifier`.

**Comment by camwebb** [comment]
>  just to clarify, are you not seeing a recordNumber column in your data.tsv export file

@kleintom sorry for the delay. I _do_ now see `recordNumber`, with data, in the DwC export file :+1:. 

It would still be very helpful to see `recordNumber` in the DwC dashboard. `fieldNumber` is there, but not `recordNumber`.
